src/main/host/memory_manager/memory_copier.rs

```diff
@@ -215,16 +215,22 @@ impl MemoryCopier {
             len: towrite,
         }];
 
-        // While the documentation for process_vm_writev says to use the pid, in
+        // While the documentation for process_vm_readv says to use the pid, in
         // practice it needs to be the tid of a still-running thread. i.e. using the
         // pid after the thread group leader has exited will fail.
-        let active_tid = Worker::active_thread_native_tid().unwrap();
-        let active_pid = Worker::active_process_native_pid().unwrap();
-
-        // Don't access another process's memory.
-        assert_eq!(active_pid, self.pid);
+        let tid = Worker::with_active_host(|host| {
+            Worker::with_active_process(|process| {
+                // Don't access another process's memory.
+                assert_eq!(process.native_pid(), self.pid);
+                let thread = process.first_live_thread_borrow(host.root()).unwrap();
+                let thread = thread.borrow(host.root());
+                thread.native_tid()
```

Comment:
Does shadow require that threads share memory? My understanding is that Linux doesn't really have threads, just processes that may or may not have shared resources (depending on the arguments to `clone`) and can be grouped into "thread groups" if `CLONE_THREAD` is set. In shadow's `clone` handler, do we require that `CLONE_VM` is always set? And if so, maybe document in shadow's `clone` handler that we require it for the memory copier to work, so that we don't accidentally relax that requirement later and break the memory copier?

Reply:
Yes, shadow currently requires `CLONE_THREAD`; and in both Linux and shadow if `CLONE_THREAD` is set, then `CLONE_SIGHAND` and `CLONE_VM` must also be set.

Added a comment for the `memory_manager` field in `RunningProcess`